core/src/main/scala/com/lightbend/paradox/markdown/Directive.scala

```diff
@@ -393,7 +393,7 @@ case class SnipDirective(page: Page, variables: Map[String, String])
     try {
       val labels = node.attributes.values("identifier").asScala
       val source = resolvedSource(node, page)
-      val filterLabels = node.attributes.booleanValue("filterLabels", variables.get("snip.filterLabels").forall(_ == "true"))
+      val filterLabels = labels.nonEmpty && node.attributes.booleanValue("filterLabels", variables.get("snip.filterLabels").forall(_ == "true"))
```

Comment:
Could we allow filtering out the labels when including a full file? Like:
* if a `filterLabels` attribute is present, use it
* if not, and a `snip.filterLabels` default it set, use it
* else, `labels.nonEmpty`
?

(do you have a use for `snip.filterLabels`? might be simpler without that option)

Reply:
I made the local attribute take precedence and if nothing is defined `labels.nonEmpty` and the attribute control it.
(The global setting was there before, I didn't want to drop it.)